libpod/options.go

```diff
@@ -669,6 +669,18 @@ func WithUser(user string) CtrCreateOption {
 	}
 }
 
+// WithDefinedUser informs us whether or not we have explicitly defined a user or if podman is using the default user
+func WithDefinedUser() CtrCreateOption {
+	return func(ctr *Container) error {
+		if ctr.valid {
+			return define.ErrCtrFinalized
+		}
+
+		ctr.config.DefinedUser = true
+		return nil
+	}
+}
```

Comment:
That would cause this change to affect podman build, right? when generating the kube we don't have access to the image file just the container we are extracting info from see libpod/kube.go L855

Reply:
Have a look at https://github.com/containers/podman/pull/11704/files